src/core/org/apache/jmeter/threads/AbstractThreadGroup.java

```diff
@@ -251,6 +251,8 @@ public boolean getOnErrorStopTestNow() {
 
     public abstract void start(int groupCount, ListenerNotifier notifier, ListedHashTree threadGroupTree, StandardJMeterEngine engine);
 
+    public abstract void addNewThread(int delay, StandardJMeterEngine engine);
```

Comment:
Is it possible that, rather than void, this could return... say, the threadId?  Having the option to keep track of the threads you've started in this manner would be useful if, for example, you want to remove them (because removing a thread requires that threadId).  The alternative is making a comparable removeOldThread() method to complete the API, but I'm sort of up in the air on whether it would be a good idea to grab a random thread and tell it to end in current JMeter (it makes perfect sense in a hypothetical future dispatch-oriented design, though).